pkg/operator/collection.go

```diff
@@ -233,22 +233,14 @@ func (r *collectionReconciler) ensureCollectorDaemonSet(ctx context.Context, spe
 		return err
 	}
 
-	var flags []string
-	// Populate export flags for collector if necessary.
-	for _, matcher := range spec.Filter.MatchOneOf {
-		flags = append(flags, fmt.Sprintf("--export.match=%q", matcher))
-	}
-	if spec.Credentials != nil {
-		p := path.Join(secretsDir, pathForSelector(r.opts.PublicNamespace, &monitoringv1.SecretOrConfigMap{Secret: spec.Credentials}))
-		flags = append(flags, fmt.Sprintf("--export.credentials-file=%q", p))
-	}
+	setContainerExtraArgs(ds.Spec.Template.Spec.Containers, CollectorPrometheusContainerName, "")
```

Comment:
Yes we want to make it explicitly empty. The problem is that customers will permanently have EXTRA_ARGs set without this (or until there's a DaemonSet update).

We continue supporting EXTRA_ARGs on Prometheus in addition to configurations for backwards-compatibility, so if you have both set, behavior may be misleading.

Reply:
> We continue supporting EXTRA_ARGs on Prometheus in addition to configurations for backwards-compatibility, so if you have both set, behavior may be misleading.

Why not atomic change with GMP version? (operator + new perms + collector failing on extra args?). I think for self install logic, we might want to consider safe route for EXTRA_ARGS (I think it's fine to fail on those from certain fork version), but def for managed we could switch atomically, no? What am I missing?

I guess the plan was to best-effort (A) get this running (use no RBAC needed form, except reset part) for a bit to reset their envvars, then  (B) remove this UPDATE and RBAC perms. Plus try to have (A) to be on all clusters to reset envvars properly (month? all regular channel?)

Long term we will have to NOT only ask for flags / vars but also for configmap to debug ANYWAY, so why not avoiding A phase?

Reply:
That's a good idea and thinking through, it should work. Do you want it 1 PR or would merging each PR separately still be okay? The PR to remove the UPDATE perms is available. If you merge it now, it will merge into this branch.